features/promotion/managing_coupons/coupon_generate_validation.feature

```diff
@@ -14,8 +14,31 @@ Feature: Coupon generate instruction validation
     Scenario: Trying to generate a new coupons without specifying its amount
         Given I want to generate a new coupons for this promotion
         When I do not specify its amount
+        And I specify its code length as 6
         And I limit generated coupons usage to 25 times
         And I make generated coupons valid until "26.03.2017"
         And I try to generate it
         Then I should be notified that generate amount is required
         And there should be 0 coupon related to this promotion
+
+    @ui
+    Scenario: Trying to generate a new coupons without specifying its code length
+        Given I want to generate a new coupons for this promotion
+        When I do not specify its code length
+        And I specify its amount as 4
+        And I limit generated coupons usage to 25 times
+        And I make generated coupons valid until "26.03.2017"
+        And I try to generate it
+        Then I should be notified that generate code length is required
+        And there should be 0 coupon related to this promotion
+
+    @ui
+    Scenario: Trying to generate a new coupons with amount and code length impossible to generate
+        Given I want to generate a new coupons for this promotion
+        When I specify its code length as 1
+        And I specify its amount as 20
+        And I limit generated coupons usage to 25 times
+        And I make generated coupons valid until "26.03.2017"
+        And I try to generate it
+        Then I should be notified that generating 20 coupons with 1 code length is not possible
```

Comment:
`Then I should be notified that generating 20 coupons with code length equal to 1 is not possible`